app/Models/Chat/Channel.php

```diff
@@ -109,18 +109,18 @@ public function receiveMessage(User $sender, string $content, bool $isAction = f
         $sentMessages->where('timestamp', '>=', Carbon::now()->subSecond($window));
 
         if ($sentMessages->count() > $limit) {
-            throw new ExcessiveChatMessagesException(trans('api.error.chat.limit_exceeded'));
+            throw new API\ExcessiveChatMessagesException(trans('api.error.chat.limit_exceeded'));
         }
 
         $message = new Message();
         $message->user_id = $sender->user_id;
         $message->content = $content;
         $message->is_action = $isAction;
+        $message->timestamp = Carbon::now();
         $message->channel()->associate($this);
         $message->save();
-        $message = $message->fresh();
 
-        $userChannel = UserChannel::where(['channel_id' => $this->channel_id, 'user_id' => $sender->user_id]);
+        $userChannel = UserChannel::where(['channel_id' => $this->channel_id, 'user_id' => $sender->user_id])->first();
```

Comment:
could just call `update` directly after `where` if it's not going to throw an exception